src/game/audio/sfx.rs

```diff
@@ -3,40 +3,40 @@ use rand::seq::SliceRandom;
 
 use crate::game::assets::{HandleMap, SfxKey};
 
-pub(super) fn play_sfx(
-    trigger: Trigger<Sfx>,
+pub(super) fn plugin(app: &mut App) {
+    app.observe(play_sfx);
+}
+
+fn play_sfx(
+    trigger: Trigger<PlaySfx>,
     mut commands: Commands,
     sfx_handles: Res<HandleMap<SfxKey>>,
 ) {
-    let event = trigger.event();
-    let source = match event {
-        Sfx::ButtonHover => &sfx_handles[&SfxKey::ButtonHover],
-        Sfx::ButtonPress => &sfx_handles[&SfxKey::ButtonPress],
-        Sfx::Step => random_step(&sfx_handles),
-    }
-    .clone_weak();
-    let settings = PlaybackSettings {
-        mode: PlaybackMode::Despawn,
-        ..default()
-    };
-    commands.spawn(AudioSourceBundle { source, settings });
+    commands.spawn(AudioSourceBundle {
+        source: sfx_handles[&match trigger.event() {
+            PlaySfx::ButtonHover => SfxKey::ButtonHover,
+            PlaySfx::ButtonPress => SfxKey::ButtonPress,
+            PlaySfx::Step => random_step(),
+        }]
```

Comment:
Lots happening in a very condensed space. Could you split this into multiple variables and steps?

Reply:
Split getting the key out into an earlier step.